runtime/stdlib.js

```diff
@@ -131,7 +131,16 @@ var caml_named_values = {};
 //Provides: caml_register_named_value (const,mutable)
 //Requires: caml_named_values, caml_jsbytes_of_string
 function caml_register_named_value(nm,v) {
-  caml_named_values[caml_jsbytes_of_string(nm)] = v;
+  nm = caml_jsbytes_of_string(nm);
+  if (!caml_named_values.hasOwnProperty(nm)) caml_named_values[nm] = v;
+  return 0;
+}
+
+//Provides: caml_unregister_named_value (const)
+//Requires: caml_named_values, caml_jsbytes_of_string
+function caml_unregister_named_value(nm) {
```

Comment:
This function doesn't have to live in the main runtime and could just live in the test directory. These tests are the only users of such primitive.

Also, you need to implement the c counter-part if you want the byte version to work.

Reply:
Something like this should work:
```C
value caml_unregister_named_value(value nm) {
   return caml_register_named_value(nm, (value) NULL);
}